index.js

```diff
@@ -79,6 +79,14 @@ class Client {
     const socket = net.connect(this.port, this.hostname)
     socket.setTimeout(options.timeout || this.timeout)
 
+    socket.on('timeout', () => {
+      socket.end()
+      const err = new Error('Request timed out')
+      err.code = 'ETIMEDOUT'
+
+      fn(err)
+    })
+
     socket
       .on('error', fn)
```

Comment:
non-blocking: Should we move the `on('timeout')` to here, with the other handlers?

Reply:
Done!